payments/webhooks.py

```diff
@@ -275,10 +276,11 @@ def set_workspace_subscription(
             new_sub = Subscription()
 
         new_sub.plan = plan.db_value
-        new_sub.plan_tier_key = plan_tier_key
+        new_sub.amount = amount or plan.credits
+        new_sub.charged_amount = charged_amount or (plan.monthly_charge * 100)
```

Comment:
i think it makes sense to rely explicitly on `amount` and `charged_amount` (vs the fallback on `plan.credits` / `plan.monthly_charge`) so we don't accidentally use wrong values for a tiered plan.